# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1344183088 -3600
#      Sun Aug 05 17:11:28 2012 +0100
# Branch gui
# Node ID 9300ee32d6ff56d448f954fd34b6d6298c9bcfbb
# Parent  acb7c0e6c317865f083bef2f92c014287ab13c51
Be more conservative during "mkae clean"

* gui/src/Makefile.am (SRCDIRS): Add ../qterminal and re-order.
(clean-local): Use rmdir instead of "rm -rf".
(CLEAN_FILES): Add libqterminal_la_MOC.

diff --git a/gui/src/Makefile.am b/gui/src/Makefile.am
--- a/gui/src/Makefile.am
+++ b/gui/src/Makefile.am
@@ -1,16 +1,16 @@
 define MOC_template =
 $(1)/moc_%.cpp: $(1)/%.h
 	test -d $(1) || mkdir -p $(1)
 	@MOC@ -o$$@ $$(DEFS) $$(DEFAULT_INCLUDES) $$(INCLUDES) $$(AM_CPPFLAGS) $$(CPPFLAGS) $$(octave_gui_CPPFLAGS) $$<
 endef
 
 SRCDIRS = m-editor qtinfo octave-adapter \
-	  ../qterminal/libqterminal ../qterminal/libqterminal/unix ../qterminal/libqterminal/win32
+	  ../qterminal/libqterminal/unix ../qterminal/libqterminal/win32 ../qterminal/libqterminal ../qterminal
 $(foreach dir,. $(SRCDIRS),$(eval $(call MOC_template,$(dir))))
 
 ui_%.h: %.ui
 	@UIC@ -o $@ $<
 
 bin_PROGRAMS = octave-gui
 
 octave_gui_MOC = \
@@ -107,17 +107,17 @@ noinst_HEADERS = \
 		 octave-adapter/octave-main-thread.h \
 		 qtinfo/parser.h \
 		 qtinfo/webinfo.h
 
 CLEANFILES = $(octave_gui_MOC) $(octave_gui_UI_H)
 
 clean-local:
 	-if test "$(srcdir)" != "."; then \
-		for d in $(SRCDIRS); do rm -rf $$d; done \
+		for d in $(SRCDIRS); do test -d $$d && rmdir $$d; done \
 	 fi
 
 BUILT_SOURCES = $(octave_gui_UI_H)
 
 EXTRA_DIST = $(octave_gui_UI)
 
 noinst_LTLIBRARIES = libqterminal.la
 
@@ -184,8 +184,10 @@ libqterminal_la_MOC += \
 		       ../qterminal/libqterminal/unix/moc_Filter.cpp \
 		       ../qterminal/libqterminal/unix/moc_QUnixTerminalImpl.cpp \
 		       ../qterminal/libqterminal/unix/moc_ScreenWindow.cpp \
 		       ../qterminal/libqterminal/unix/moc_SelfListener.cpp \
 		       ../qterminal/libqterminal/unix/moc_TerminalModel.cpp \
 		       ../qterminal/libqterminal/unix/moc_TerminalView.cpp \
 		       ../qterminal/libqterminal/unix/moc_Vt102Emulation.cpp
 endif
+
+CLEANFILES += $(libqterminal_la_MOC)
